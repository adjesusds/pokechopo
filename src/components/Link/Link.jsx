// External Packages
import { node, oneOf, string } from 'prop-types'
import React from 'react'

// Internal Packages
import StyledLink from './styled/Link'

/**
 * Creates a new Link component.
 * @class
 */
const Link = ({ children, href, target }) => (
  <StyledLink href={href} target={`_${target}`}>
    {children}
  </StyledLink>
)

/** @override */
Link.propTypes = {
  children: node.isRequired,
  href: string.isRequired,
  target: oneOf(['blank', 'parent', 'self', 'top']),
}

/** @override */
Link.defaultProps = {
  target: 'self',
}

export default Link
