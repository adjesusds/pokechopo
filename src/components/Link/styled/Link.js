// External Packages
import styled from 'styled-components'

// Internal Packages
import { colors, durations } from 'config/palette'

/**
 * @constant {Element} style The style of the component.
 */
const style = styled.a`
  color: ${colors.white};
  text-decoration-skip: ink;
  transition: color ${durations.fast} ease-in;

  &:hover {
    color: ${colors.yellow.x500};
  }
`

export default style
