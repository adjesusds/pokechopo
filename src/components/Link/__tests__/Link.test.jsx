// External Packages
import { shallow } from 'enzyme'
import React from 'react'

// Internal Packages
import Link from '../Link'

describe('<Link />', () => {
  it('should render the component on valid props', () => {
    let component

    component = shallow(<Link href="https://pokeapi.co/api/v2/pokemon/1/">Hello, World!</Link>)

    expect(component).toMatchSnapshot()

    component = shallow(
      <Link href="https://pokeapi.co/api/v2/pokemon/1/" target="blank">
        Hello, World!
      </Link>
    )

    expect(component).toMatchSnapshot()

    component = shallow(
      <Link href="https://pokeapi.co/api/v2/pokemon/1/" target="parent">
        Hello, World!
      </Link>
    )

    expect(component).toMatchSnapshot()

    component = shallow(
      <Link href="https://pokeapi.co/api/v2/pokemon/1/" target="top">
        Hello, World!
      </Link>
    )

    expect(component).toMatchSnapshot()
  })
})
