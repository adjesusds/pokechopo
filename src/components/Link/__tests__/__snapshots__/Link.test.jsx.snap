// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> should render the component on valid props 1`] = `
<styled.a
  href="https://pokeapi.co/api/v2/pokemon/1/"
  target="_self"
>
  Hello, World!
</styled.a>
`;

exports[`<Link /> should render the component on valid props 2`] = `
<styled.a
  href="https://pokeapi.co/api/v2/pokemon/1/"
  target="_blank"
>
  Hello, World!
</styled.a>
`;

exports[`<Link /> should render the component on valid props 3`] = `
<styled.a
  href="https://pokeapi.co/api/v2/pokemon/1/"
  target="_parent"
>
  Hello, World!
</styled.a>
`;

exports[`<Link /> should render the component on valid props 4`] = `
<styled.a
  href="https://pokeapi.co/api/v2/pokemon/1/"
  target="_top"
>
  Hello, World!
</styled.a>
`;
